src/Features/CSharp/Portable/Completion/KeywordRecommenders/AbstractSpecialTypePreselectingKeywordRecommender.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.KeywordRecommenders
+{
+    internal abstract class AbstractSpecialTypePreselectingKeywordRecommender : AbstractSyntacticSingleKeywordRecommender
+    {
+        public AbstractSpecialTypePreselectingKeywordRecommender(
+            SyntaxKind keywordKind, 
+            bool isValidInPreprocessorContext = false, 
```

Comment:
i personally hate optional parameters in our internal APIs.  Because now it's not clear to me at the callsite that we're depending on default values, and it's too simple too accidently get the wrong behavior.  Can you make these required?